server/src/main/java/io/crate/execution/dml/IndexDocumentBuilder.java

```diff
@@ -191,6 +192,8 @@ public ParsedDocument build(String id) {
         BytesRef idBytes = Uid.encodeId(id);
         addField(new Field(SysColumns.Names.ID, idBytes, SysColumns.ID.FIELD_TYPE));
 
+        addField(new BinaryDocValuesField(SysColumns.Names.ID_BYTES, idBytes));
```

Comment:
Any reason to use a different field instead of using a different fieldType with the `_id` field?

Also - given that the read is behind a version check, shouldn't this be behind a version check too, to avoid writing the new dvField in old tables where it will never be used?

Reply:
Annoyingly, the version here in IndexDocumentBuilder is the shard version, while the version needed to correctly do this is the table version, due to complications in ReservoirSampler.  I'm not sure how complex it would be to fix things in the sampler - I can take a quick look and see if its a simple fix and do that first?

I used a different field as I was getting weird errors in tests about changing the DV implementation on a field, but looking again I think I had missed a change to the definition of Tombstone docs.  Will update.